### PR TITLE
Understanding Computer network services

### LINKED ISSUE #4993: Understanding computer network services
#### Proposed article introduction
Network service is an operand in the network provided by a server. The application runs at the network
application layer and above, and include; IP addressing, Domain Name System (DNS), Internet access,
security products (firewalls), primary domain service, web content filtering, VPN termination and
intrusion prevention systems (IPS), and the tools to support these services.
A server component for each service is usually run on one or more computers which is accessed via a network by client components running on other devices. Clients and servers often have a user interface
and sometimes other hardware associated with it.
In this context we are going to cover :
- Basic Network Design Requirements
- Scaling a network
- Network services and their functions (DHCP, DNS, File Server Services, Printing, Authentication,
etc)
- Automatic Network Services
- Configuring Network Services.

#### Key takeaways 
    i. The reader should understand the basics in designing a computer network.
    ii. The reader should be able to figure out factors to consider when scaling a network.
    iii. The reader should be able to understand the different network services and their functions.
    iv. The reader should be able to understand configuration of the network services
#### Article quality
This article will provide an in-depth tutorial on how a network can be developed from scratch and the services that will be provided by the network. Unlike other articles, this article will explain definitively the concepts surrounding network authentication, file server services and DHCP.  In addition, it will show illustratory diagrams clearly explaining the concept in question.
#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by gorleroy**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #4993

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Plagiarism check is done.

File: content/articles/computer-network-services/index.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+- Routing service
+1. This can be used in place of routers.
+
+2. It allows a server with two network interface cards installed to route data between networks or between a network (private) and the internet.
+
+3.  Based on the manner of segmentation, different IP addresses are assigned to the two [network interface cards](https://www.tutorialspoint.com/what-is-network-interface-card-nic) on the network.
+
+### Automation of Network Services
+
+It is an approach to manage network services and resources That is; automating configuration, testing, maintenance, provision, management, and operation of a network and its’ devices both virtual and physical.
```
Comment (mbaabuones) [inline content/articles/computer-network-services/index.md:165]: Revise this sentence. Please avoid such errors.

File: content/articles/computer-network-services/index.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+| Hostname      | Select a good and descriptive hostname with characters, not exceeding 30.| 
+| Primary DNS   | Indicate primary DNS server IP address.|
+| Secondary DNS | Indicate primary DNS server IP address.|
+| DNS Domain| Indicate a list of default domains. Use commas to separate the domains|
+| Preferred DNS Response| This indicates the preferred requests and responses by the DNS.  Pick ‘V4’ if only IPv4 is needed. Select ‘Both’ if both IPv4 and IPv6 are needed.
+| Bandwidth management | This is used for bandwidth management only.|
+| Maximum Bandwidth| It Specifies the maximum bandwidth for all traffic in the network.|
+| Tunnels Maximum | It Specifies the maximum bandwidth for traffic in the VPN. |
+| Bandwidth | The total maximum bandwidth must be higher than the value of VPN  maximum bandwidth. |
+
```
Comment (mbaabuones) [inline content/articles/computer-network-services/index.md:193]: Introduce a section for conclusion. In this section, provide a summary of what the article has covered and the concluding remarks. You can write two paragraphs.

File: content/articles/computer-network-services/index.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+- Services clustering
+- RAID
+
+4. Quality of Services (QoS)
+
+QoS is a metric measuring the level of distribution of the original product in regards to the quality of the
+transmitted data despite the distance. It can be achieved through data prioritization for devices with the
+capability.
+
+### Scaling a Network
```
Comment (mbaabuones) [inline content/articles/computer-network-services/index.md:65]: write all headers in the required format. Please refer to other published articles.

**Review by mbaabuones — CHANGES_REQUESTED** [review]
@gorleroy Thank you for submitting your first article. I take this opportunity to welcome you to the Section Engineering program. Your article is informative and easy to understand. However, the article needs some changes to meet the threshold for publication. Please go through the following points keenly and revise the article accordingly. 
* As a first time contributor, you are required to upload your author bio and photo (named avatar) in the **authors** folder situated in the **contents** folder. 
* Please revisit [this folder](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) to familiarize yourself with all the contribution guidelines.
* The introduction paragraphs and conclusion of this article can be optimized. Please go through the [published articles](https://www.section.io/engineering-education/) to see how authors have structured their articles. The published articles will help you gain an overview of the desired quality. 
* There are some grammatical errors in the article. Use Grammarly.com to check and correct these errors. 
* Please address the specific comments pointed out in the article. 
This PR will be closed so that you can work on all the comments outside the queue. This will allow other ready articles to be processed. Please make sure all the suggested changes are made. When you finish working on the changes, open a new PR and indicate in the subject line that it is a duplicate. Ping me when you do this. Please note that if there are further changes needed after opening a duplicate PR, it will be closed again. To avoid this, ensure you have addressed all the highlighted issues. 

NB: Ensure you proofread the article when you finish revising it. Fix any other error, typo, or flow issue in the article.

**Pull request closed without merging**